git-branchless/src/commands/smartlog.rs

```diff
@@ -210,7 +317,23 @@ mod graph {
         }
     }
 
-    /// Construct the smartlog graph for the repo.
+    /// What kind of smartlog is desired?
+    #[derive(Debug)]
+    pub enum SmartlogVariant {
+        /// A smartlog with all commits connected back to the main branch.
+        Connected,
```

Comment:
```suggestion
        /// A smartlog with all commits connected back to the main branch.
        #[default]
        Connected,
```

You should be able to annotate the default value as long as it doesn't take any parameters: https://doc.rust-lang.org/std/default/trait.Default.html#enums

Reply:
Well, yes and no. It works just fine locally (on Rust 1.64), but clippy rejects it in CI. Apparently it wasn't [stabilized until 1.62](https://github.com/rust-lang/rust/pull/94457#issuecomment-1090989086), but CI is running 1.61.

Unless you want me to look into adding the Cargo flag (or clippy annoation?) to allow it, I'm leaning towards leaving it as is for now and then updating it later, once we're running 1.62+.